### PR TITLE
Adjust default directory for configuration file save and load

### PR DESCRIPTION
**Pull request by johahauf**
As suggested in Issue #1065 The dialog for opening a configuration file could have a better default directory

### CODE REVIEW

File: vncviewer/ServerDialog.cxx [hunk 1]
```diff
@@ -53,6 +54,23 @@ static LogWriter vlog("ServerDialog");
...
+static void updateUsedDir(const char* filename)
+{
+#ifndef  _WIN32
+  char * name = strdup(filename);
+  usedDir = strdup(dirname(name));
+  free(name);
+#else
+  char dir[_MAX_DIR], drive[_MAX_DRIVE], path[_MAX_PATH];
+  if (_splitpath_s(filename, drive, _MAX_DRIVE, dir, _MAX_DIR, NULL, 0, NULL, 0) == 0) {
+    if (_makepath_s( path, _MAX_PATH, drive, dir, NULL, NULL) == 0) {
```
Comment (CendioOssman) [inline vncviewer/ServerDialog.cxx:67]: `dirname()` seems to be provided by mingw, so I don't think we need this alternative for Windows?

File: vncviewer/ServerDialog.cxx [hunk 2]
```diff
@@ -53,6 +54,23 @@ static LogWriter vlog("ServerDialog");
 
 const char* SERVER_HISTORY="tigervnc.history";
 
+static char* usedDir = NULL;
+static void updateUsedDir(const char* filename)
```
Comment (CendioOssman) [inline vncviewer/ServerDialog.cxx:58]: Can't this be part of the class?

**Review by CendioOssman — CHANGES_REQUESTED** [review]
This looks like a nice addition, thanks! I've got a couple of notes though.

File: vncviewer/ServerDialog.cxx (hunk 1 above)
Comment (johahauf) [reply vncviewer/ServerDialog.cxx:67]: Yes, you are right, it is working. Will remove it.

**Pull request merged**